### PR TITLE
add includeTrace to runnerOpts. Includes traces object in return value.

### PR DESCRIPTION
**Pull request by paulirish**
If this config value is passed to the `Runner.run` then the big return value will include the traces artifact.

This enables something like:

<a href="https://gyazo.com/91894b27564fd54608c25b9b94697206">

</a>

➡️  View GIF: https://gyazo.com/91894b27564fd54608c25b9b94697206

### CODE REVIEW

**Comment by paulirish** [comment]
I'm open to other options if ya'll have ideas.

File: lighthouse-core/runner.js [hunk 1]
```diff
@@ -137,6 +146,7 @@ class Runner {
           initialUrl: opts.initialUrl,
           url: opts.url,
           audits: formattedAudits,
+          traces: opts.includeTrace ? tracesToInclude : undefined,
```
Comment (ebidel) [inline lighthouse-core/runner.js]: If you init'd `let tracesToInclude;` then you could just do: `traces: tracesToInclude` here.

**Comment by brendankenny** [comment]
this does pull out data from our nice pipeline of processing results into a side channel, but I'm not sure if there's a good way to avoid that if we want to hold on to artifact data. From upcoming plans that may require artifacts to live past the lifetime of a single run, I think it makes more sense to actually get `artifacts` piped through to that final stage and grab `artifacts.traces` only at the very end.

You could save it in a variable in an outer scope like you are now or do something ugly like pass through an object with `artifacts` and `auditResults` instead of just `auditResults` there at the end.

**Comment by paulirish** [comment]
@brendankenny sounds good. I'll put this PR on hold while some other bits get sorted out

**Comment by paulirish** [comment]
cc @WeiweiAtGit since you'll probably end up doing something slightly similar.

**Pull request closed without merging**